### PR TITLE
ISLANDORA-2454: Expose the search table and add a default view

### PR DESCRIPTION
**Pull request by BrandonMorr**
**NOTE: this pull supersedes https://github.com/Islandora/islandora_usage_stats/pull/68**

**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2454

# What does this Pull Request do?

Exposes the `islandora_usage_stats_top_searches` table to views so it can be used to generate a report.

# What's new?

A view showing the top searches in the repository.

# How should this be tested?

Pull the module and clear your cache/rebuild your menus. Notice there is now an `Islandora Usage Stats Top Searches` option available at `admin/reports/islandora-usage-stats-top-searches`. Clicking it shows a list of search results and the frequency of hits.

# Interested parties
@Islandora/7-x-1-x-committers

### CODE REVIEW

**Comment by bondjimbond** [comment]
@BrandonMorr The link still won't show up for me. Is it working for you?

Screen Shot 2019-07-10 at 3 25 12 PM

**Comment by BrandonMorr** [comment]
It appears to be working on my end, here's a screen grab:

Screen Shot 2019-07-10 at 4 30 21 PM

Some example search queries:
* `site.com/islandora/search/catch_all_fields_mt%3A(test)`
* `site.com/islandora/search/test`
* `site.com/islandora/search/mods_subject_topic_mt%3A(big\ test)`

Have you tried disabling/re-enabling the module?

If you go to the View's configuration UI (`/admin/structure/views/view/usage_searches`) does the search term field show that it should be rewritten as a link?

**Comment by bondjimbond** [comment]
> Have you tried disabling/re-enabling the module?

Yes; no change.

> If you go to the View's configuration UI (/admin/structure/views/view/usage_searches) does the search term field show that it should be rewritten as a link?

No links...

Screen Shot 2019-07-10 at 3 59 59 PM

**Comment by BrandonMorr** [comment]
Very strange, you're absolutely certain that this line is present in the code you have checked out?

https://github.com/Islandora/islandora_usage_stats/pull/70/files#diff-917994de0b7acc05b7b03bd714d3877eR549

**Comment by bondjimbond** [comment]
Yep!

Screen Shot 2019-07-10 at 4 09 01 PM

**Comment by BrandonMorr** [comment]
Weird!! From the view's configuration UI, do you see this option enabled for the `Search Term` field?

Screen Shot 2019-07-10 at 5 45 26 PM

Apologies for the crazy line of questioning, I just have no clue how you're not seeing the links!

**Comment by bondjimbond** [comment]
Nope, no link configured. Box not checked.

**Comment by bondjimbond** [comment]
I would try uninstalling islandora_usage_stats, except there is no uninstall option.

**Comment by BrandonMorr** [comment]
Hi Brandon, not sure what you mean by no uninstall option. If you mean the checkbox is greyed out in the UI, that means there is a module that is dependent on Islandora Usage Stats.

If you have drush installed on your testing environment, you could try this out:
```
drush dis islandora_usage_stats
drush pm-uninstall islandora_usage_stats
drush en islandora_usage_stats
```

You didn't happen to modify the View did you? I believe once there are changes between what is in code and what exists in the db it won't listen for any changes in the View's definition.

From `/admin/structure/views` do you see _In Code_ displayed for the View:

Screen Shot 2019-07-11 at 11 44 10 AM

**Comment by bondjimbond** [comment]
> not sure what you mean by no uninstall option. If you mean the checkbox is greyed out in the UI, that means there is a module that is dependent on Islandora Usage Stats.

No problem disabling the modules; I understand dependencies. I'm saying that once the module is disabled, it doesn't appear in the "Uninstall" menu. I'll try doing it by drush though.

> You didn't happen to modify the View did you

Didn't touch it.

> From /admin/structure/views do you see In Code displayed for the View

Yes.

**Comment by bondjimbond** [comment]
Update: Disabled everything, uninstalled everything, re-enabled, ran some searches.

Still no links in the View.

**Comment by bondjimbond** [comment]
@BrandonMorr I wonder if your local version might be affected by local changes during development? What happens if you delete the view, uninstall islandora_usage_stats, and re-enable? Do you still get the links?

**Comment by BrandonMorr** [comment]
> @BrandonMorr I wonder if your local version might be affected by local changes during development? What happens if you delete the view, uninstall islandora_usage_stats, and re-enable? Do you still get the links?

For what it's worth, I didn't do any of the development for this work. I just nuked the `dsm()` call and pushed it up to my fork of the module.

I'm truly stumped... I just spun up another test environment, pulled down the changes in this pull, made a test search and surely enough:

Screen Shot 2019-07-11 at 1 08 36 PM

Curious, are you testing these changes in a multisite environment? Have you run something like:
```
drush @sites cc all
```

**Comment by BrandonMorr** [comment]
Just a thought, it might be useful to see if any of the other committers could quickly give this a smoke test. Perhaps @bryjbrown?

**Comment by bryjbrown** [comment]
@BrandonMorr I was thinking the same thing, trying to get this running now.

**Comment by bondjimbond** [comment]
> Curious, are you testing these changes in a multisite environment? 

Nope, single site (Vagrant machine). I can't fathom why this won't work. But if @bryjbrown gives it a pass, I'll be happy enough with it and accept that something weird happened.

**Comment by bryjbrown** [comment]
@BrandonMorr So, on a fresh machine, I disabled the module, cleared the cache, pulled and switched to this PR, enabled the module, cleared the cache again, then checked the report. The items in the "Search Terms" field are linked for me.

**Comment by bondjimbond** [comment]
Super weird. Glad it works, though. @bryjbrown want to give an approval, then?

**Comment by bryjbrown** [comment]
@bondjimbond Yeah I can, but before I do I just want clarification: you said in https://github.com/Islandora/islandora_usage_stats/pull/70#issuecomment-510509431 that the box wasn't checked. Is the issue here for you that the "Output this field as a link" box was not checked by default, but after checking it they search terms became linked, or is the issue that the search terms are unlinked even after checking the box? I think the first situation is acceptable because, worst case scenario, we're defaulting to the original behavior, but the second situation warrants a closer look perhaps.

**Comment by bondjimbond** [comment]
> Is the issue here for you that the "Output this field as a link" box was not checked by default, but after checking it they search terms became linked, or is the issue that the search terms are unlinked even after checking the box?

The box is not checked, and if you do check it, there is no data there. I'm sure it would work if I added my own linking pattern, but that's not what we want. In my box it looks as if there was never any kind of linking behaviour defined in the code at all.

**Comment by bryjbrown** [comment]
@bondjimbond Thats really strange. I'd play around with the config, but for some extremely odd reason my absolutely fresh Vagrant VM isn't showing a Views tab under Structure at all, and when I try to manually hit `/admin/structure/views` it drops me to what I'd see at `/admin/strucutre`, as if I don't have permission even though I checked and I absolutely do. @BrandonMorr any thoughts?

**Comment by bondjimbond** [comment]
@bryjbrown Just to check the obvious thing, is Views UI enabled?

**Comment by bryjbrown** [comment]
@bondjimbond ... no it wasn't.
whoopsie

**Comment by bryjbrown** [comment]
@bondjimbond I've been working on Drupal 8 stuff so much lately I forgot Views aren't first class entities in 7.x. MY BAD Y'ALL.

**Comment by bryjbrown** [comment]
So, now that I can see the Views UI, I can also confirm that checking/unchecking the "Output this field as a link" box seems to have the intended effect. I'm sold.

**Comment by BrandonMorr** [comment]
Wicked. Are we good to start the clock, then?

**Review by bryjbrown — APPROVED** [review]

**Comment by bryjbrown** [comment]
@BrandonMorr Yup. :clock1:

**Comment by BrandonMorr** [comment]
Hey @bryjbrown, are we good to merge?

**Comment by bryjbrown** [comment]
@BrandonMorr Yep, so sorry, this is what happens when I forget to set calendar reminders :face_with_head_bandage:

**Comment by BrandonMorr** [comment]
No worries at all :-), thanks fellas!

**Pull request merged**
